titiler/pgstac/reader.py

```diff
@@ -0,0 +1,74 @@
+"""Custom STAC reader."""
+
+from typing import Dict, Optional, Set, Type
+
+import attr
+import pystac
+from morecantile import TileMatrixSet
+from rasterio.crs import CRS
+from rio_tiler.constants import WEB_MERCATOR_TMS, WGS84_CRS
+from rio_tiler.errors import InvalidAssetName, MissingAssets
+from rio_tiler.io import BaseReader, COGReader, MultiBaseReader
+from rio_tiler.io.stac import DEFAULT_VALID_TYPE, _get_assets
+
+
+@attr.s
+class PgSTACReader(MultiBaseReader):
+    """Custom STAC Reader."""
```

Comment:
Custom version of rio-tiler STACReader but takes **only** STAC Item as input (instead of url or item)